src/racer/core.rs

```diff
@@ -955,6 +956,25 @@ fn complete_from_file_(
             let is_global = expr.starts_with("::");
             let is_use = line.starts_with("use ");
 
+            // when in the function ident position, only look for methods
+            // from a trait to complete.
+            if util::in_fn_name(line) {
+                trace!("Path is in fn declaration: `{}`", expr);
+
+                for m in nameres::resolve_method(
```

Comment:
The formatting on this is really hard to parse. It might be better to assign the returned Iter/IntoIter to a temporary variable first. Eg

```rust
let methods = nameres::resolve(/* ... */);
for m in methods {
    out.push(m);
}
```

Actually, maybe it would be better if resolve_method just returned a `Vec` instead of an iterator so that it may be appended directly only `out` instead of doing this iterative thing.